Update winrt-js-conversion.md for out array params (#3182)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/how-to/winrt-js-conversion.md

| Class static member | JavaScript object property | See below. |
| Class constructor | JavaScript constructor and function | See below. |

> A couple notes for passing JavaScript objects to host objects:

Reviewer: ```suggestion
When passing JavaScript objects to host objects:
```

> *  If JavaScript Date objects need to be passed to host objects as `VT_DATE`, the host objects property `shouldSerializeDates` should be set to true. By default, it is passed to host as string using `JSON.stringify`.

Reviewer: ```suggestion
*  If JavaScript `Date` objects need to be passed to host objects as `VT_DATE`, the host object's `shouldSerializeDates` property should be set to `true`. By default, `Date` objects are passed to the host as `string`, by using `JSON.stringify`.
```
* it - check whether "it" means Date objects.  prefer being specific instead of "it".

> *  If JavaScript typed arrays need to be passed to host objects as array, the host objects property `shouldPassTypedArraysAsArrays` should be set to true. By default, it is passed to host as `IDispatch`.

Reviewer: ```suggestion
*  If JavaScript typed arrays need to be passed to host objects as `array`, set the host object's `shouldPassTypedArraysAsArrays` property to `true`. By default, typed arrays are passed to the host as `IDispatch`.
```
* should - corp-level style guide re: "should": https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/should-vs-must
* it - check whether "it" means typed arrays.  prefer being specific instead of "it".

console.assert(result.intParam3 == 2);
```

> For array type `out` parameters, the array will need to be passed into the method's parameter list when calling from JavaScript. For non-`void` return type, the result array will replace the array passed in for the method call. For `void` return type, the result array will be the result of the method call.

Reviewer: ```suggestion
For array type `out` parameters, the array will need to be passed into the method's parameter list when calling from JavaScript. For a non-`void` return type, the result array will replace the array that's passed in for the method call. For `void` return type, the result array will be the result of the method call.
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which - "Include _that_ even if the sentence is clear without it. It helps to clarify the sentence structure."

```cpp
// Both methods update input array values to index values
String NonVoidMethodWithArrayOutParam(out Int[] intArrayParam);

Void VoidMethodWithArrayOutParam(out Int[] intArrayParam);
```

```javascript
let input_array1 = [0, 0, 0];

let result1 = object.NonVoidMethodWithArrayOutParam(input_array1);

console.assert(input_array1 == [0, 1, 2])

let input_array2 = [0, 0, 0];

let result2 = object.VoidMethodWithArrayOutParam(input_array2);

console.assert(result2 == [0, 1, 2]);
```

> If passing typed arrays as array `out` parameters, `chrome.webview.hostObjects.options.shouldPassTypedArraysAsArrays` will need to be set to true.

Reviewer: ```suggestion
If passing typed arrays as array `out` parameters, `chrome.webview.hostObjects.options.shouldPassTypedArraysAsArrays` needs to be set to `true`.
```
favor present tense if possible, per corp-level style guide

> If passing typed arrays as array `out` parameters, `chrome.webview.hostObjects.options.shouldPassTypedArraysAsArrays` will need to be set to true.

Reviewer: ```suggestion
If passing typed arrays as array `out` parameters, `chrome.webview.hostObjects.options.shouldPassTypedArraysAsArrays` needs to be set to `true`.
```
favor present tense if possible, per corp-level style guide
